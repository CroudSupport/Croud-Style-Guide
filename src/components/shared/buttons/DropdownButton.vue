
<template>
    <div class="ui floating dropdown icon button" ref="dropdown" :class="{disabled: loading}">
        <i class="dropdown icon" ></i>
        <div class="menu">
            <div v-for="item in menu" class="item" @click="item.action" :key="item.title">{{ item.title }}</div>
        </div>
    </div>
</template>

<script>
    /**
    * Dropdown button atom that is designed to be used in a group of buttons to add more options to the main button
    *
    * @example ./croud-dropdown-button.md
    */
    export default {
        name: 'croud-dropdown-button',

        props: {
            /*
            * Array of menu item objects in the format {title: foo, action() {console.log()}}
            */
            menu: {
                type: Array,
                default: () => [],
            },

            /*
            * Loading flag to disable element
            */
            loading: {
                type: Boolean,
                default: false,
            },
        },

        mounted() {
            $(this.$refs.dropdown).dropdown({
                action: 'hide',
            })
        },
    }
</script>

<style scoped>

</style>

