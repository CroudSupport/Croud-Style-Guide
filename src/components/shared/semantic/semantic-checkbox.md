### semantic-checkbox
For our checkboxes, we use the semantic-checkbox component from the [croud-vue-semantic](https://github.com/CroudSupport/vue-semantic) library. Below are a few examples of semantic-checkbox in action. Click [here](http://croudsupport.github.io/vue-semantic/#semantic-checkbox) for full component docs.

### Basic usage

    <semantic-checkbox v-model="checkbox" label="🍕 Pizza" />

### Styling
Pass through the **type** prop to change the style of checkbo

    <div class="ui form">
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Checkbox" />
        </div>
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Slider" type="slider"/>
        </div>
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Toggle" type="toggle" />
        </div>
    </div>

### Disabled

    <semantic-checkbox label="🍕 Pizza" type="toggle" :disabled="true"/>
