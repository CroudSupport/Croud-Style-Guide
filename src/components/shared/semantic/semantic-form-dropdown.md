### semantic-form-dropdown
For our form dropdowns, we use the semantic-form-dropdown component from the [croud-vue-semantic](https://github.com/croudtech/vue-semantic) library. Below are a few examples of semantic-form-dropdown in action. Click [here](http://croudtech.github.io/vue-semantic/#semantic-form-dropdown) for full component docs.

### Basic usage

    <semantic-form-dropdown v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]"></semantic-form-dropdown>

### Styling
You can use the props to change the styling of the dropdown

    <div class="ui form">
        <div class="field">
            <label>Fluid</label>
            <semantic-form-dropdown fluid v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
        </div>

        <div class="three fields">
            <div class="field">
                <label>Search</label>
                <semantic-form-dropdown search v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
            <div class="field">
                <label>Multiple</label>
                <semantic-form-dropdown multiple v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
            <div class="field">
                <label>Disabled</label>
                <semantic-form-dropdown :disabled="true" v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
        </div>

        <div class="inline field">
            <label>Inline</label>
            <semantic-form-dropdown :inline="true" v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
        </div>
    </div>

### Using slots
The **default** slot allows you to add custom options to the top of the dropdown

    <semantic-form-dropdown v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]">
        <div class="item" v-data="3">
            <b>Baz</b>
        </div>
    </semantic-form-dropdown>

### Handle Events
This component emits the **dropdown-selected** event when an item is selected.

    <semantic-form-dropdown :value="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" @dropdown-selected="$toasted.success('TaDa!!!')"></semantic-form-dropdown>
