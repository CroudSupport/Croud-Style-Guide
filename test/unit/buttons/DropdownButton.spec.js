
import Vue from 'vue'
import DropdownButton from '../../../src/components/shared/buttons/DropdownButton'
import '../../../semantic/dist/semantic'


const Constructor = Vue.extend(DropdownButton)

const vm = new Constructor({
    propsData: {
        menu: [
            { title: 'foo', action: () => console.log('foo') },
            { title: 'bar', action: () => console.log('bar') },
        ],
        loading: false,
    },
}).$mount()

describe('DropdownButton', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })

    it('should disable the dropdown if loading', (done) => {
        vm.loading = true
        vm.$nextTick(() => {
            expect(vm.$el).toMatchSnapshot()
            done()
        })
    })
})
