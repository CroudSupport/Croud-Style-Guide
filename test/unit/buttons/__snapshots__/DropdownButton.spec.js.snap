// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownButton should disable the dropdown if loading 1`] = `
<div
  class="ui floating dropdown icon button disabled"
  tabindex="0"
>
  <i
    class="dropdown icon"
  />
  <div
    class="menu"
    tabindex="-1"
  >
    <div
      class="item"
    >
      foo
    </div>
    <div
      class="item"
    >
      bar
    </div>
  </div>
</div>
`;

exports[`DropdownButton should match the snapshot 1`] = `
<div
  class="ui floating dropdown icon button"
  tabindex="0"
>
  <i
    class="dropdown icon"
  />
  <div
    class="menu"
    tabindex="-1"
  >
    <div
      class="item"
    >
      foo
    </div>
    <div
      class="item"
    >
      bar
    </div>
  </div>
</div>
`;
